ASTRA CUDA tests not skipped if only ASTRA CPU is installed

Here are the details. I have an environment with ASTRA CPU but without CUDA.

```
==================================================== FAILURES ====================================================
___________________________________________ test_parallel_3d_projector ___________________________________________

    def test_parallel_3d_projector():
        """Create ASTRA 3D projectors."""
    
        odl.tomo.astra_projector('nearest', VOL_GEOM_3D, PROJ_GEOM_3D, ndim=3,
>                                impl='cuda')

odl/test/tomo/backends/astra_setup_test.py:284: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
odl/tomo/backends/astra_setup.py:549: in astra_projector
    return astra.projector3d.create(proj_cfg)
../../miniconda/envs/astra_cpu/lib/python3.5/site-packages/astra/projector3d.py:36: in create
    return p.create(config)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   Exception: Error creating Projector3D.

astra/projector3d_c.pyx:59: Exception
______________________________________________ test_astra_algorithm ______________________________________________

    def test_astra_algorithm():
        """Create ASTRA algorithm object."""
    
        direction = 'forward'
        ndim = 2
        impl = 'cpu'
        vol_id = odl.tomo.astra_data(VOL_GEOM_2D, 'volume', ndim=ndim)
        rec_id = odl.tomo.astra_data(VOL_GEOM_2D, 'volume', ndim=ndim)
        sino_id = odl.tomo.astra_data(PROJ_GEOM_2D, 'projection', ndim=ndim)
        proj_id = odl.tomo.astra_projector('nearest', VOL_GEOM_2D, PROJ_GEOM_2D,
                                           ndim=ndim, impl=impl)
    
        # Checks
        with pytest.raises(ValueError):
            odl.tomo.astra_algorithm('none', ndim, vol_id, sino_id, proj_id, impl)
        with pytest.raises(ValueError):
            odl.tomo.astra_algorithm(direction, 0, vol_id, sino_id, proj_id, impl)
        with pytest.raises(ValueError):
            odl.tomo.astra_algorithm('none', ndim, vol_id, sino_id, proj_id,
                                     'none')
        with pytest.raises(ValueError):
            odl.tomo.astra_algorithm('backward', ndim, vol_id, sino_id,
                                     proj_id=None, impl='cpu')
        alg_id = odl.tomo.astra_algorithm(direction, ndim, vol_id, sino_id,
                                          proj_id, impl)
        astra.algorithm.delete(alg_id)
    
        # 2D CPU
        ndim = 2
        impl = 'cpu'
        for direction in {'forward', 'backward'}:
            alg_id = odl.tomo.astra_algorithm(direction, ndim, vol_id, sino_id,
                                              proj_id, impl)
            astra.algorithm.delete(alg_id)
    
        # 2D CUDA
        proj_id = odl.tomo.astra_projector('nearest', VOL_GEOM_2D, PROJ_GEOM_2D,
>                                          ndim=ndim, impl='cuda')

odl/test/tomo/backends/astra_setup_test.py:333: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
odl/tomo/backends/astra_setup.py:547: in astra_projector
    return astra.projector.create(proj_cfg)
../../miniconda/envs/astra_cpu/lib/python3.5/site-packages/astra/projector.py:36: in create
    return p.create(config)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   Exception: Error creating projector.

astra/projector_c.pyx:63: Exception
========================= 2 failed, 2823 passed, 566 skipped, 1 xfailed in 12.05 seconds =========================
```

The failures are in `astra_setup.py` which tries to run all tests and does not skip properly.

--- a/odl/test/tomo/backends/astra_setup_test.py
+++ b/odl/test/tomo/backends/astra_setup_test.py
@@ -292,9 +292,6 @@ def test_parallel_2d_projector():
 def test_parallel_3d_projector():
     """Create ASTRA 3D projectors."""
 
-    odl.tomo.astra_projector('nearest', VOL_GEOM_3D, PROJ_GEOM_3D, ndim=3,
-                             impl='cuda')
-
     # Run as a real test once ASTRA supports this construction
     with pytest.raises(ValueError):
         odl.tomo.astra_projector('nearest', VOL_GEOM_3D, PROJ_GEOM_3D,
@@ -305,6 +302,15 @@ def test_parallel_3d_projector():
                                  ndim=3, impl='cpu')
 
 
+@pytest.mark.skipif(not odl.tomo.ASTRA_CUDA_AVAILABLE,
+                    reason="ASTRA cuda not available")
+def test_parallel_3d_projector_gpu():
+    """Create ASTRA 3D projectors on GPU."""
+
+    odl.tomo.astra_projector('nearest', VOL_GEOM_3D, PROJ_GEOM_3D, ndim=3,
+                             impl='cuda')
+
+
 def test_astra_algorithm():
     """Create ASTRA algorithm object."""
 
@@ -312,7 +318,6 @@ def test_astra_algorithm():
     ndim = 2
     impl = 'cpu'
     vol_id = odl.tomo.astra_data(VOL_GEOM_2D, 'volume', ndim=ndim)
-    rec_id = odl.tomo.astra_data(VOL_GEOM_2D, 'volume', ndim=ndim)
     sino_id = odl.tomo.astra_data(PROJ_GEOM_2D, 'projection', ndim=ndim)
     proj_id = odl.tomo.astra_projector('nearest', VOL_GEOM_2D, PROJ_GEOM_2D,
                                        ndim=ndim, impl=impl)
@@ -339,6 +344,18 @@ def test_astra_algorithm():
         alg_id = odl.tomo.astra_algorithm(direction, ndim, vol_id, sino_id,
                                           proj_id, impl)
         astra.algorithm.delete(alg_id)
+
+
+@pytest.mark.skipif(not odl.tomo.ASTRA_CUDA_AVAILABLE,
+                    reason="ASTRA cuda not available")
+def test_astra_algorithm_gpu():
+    """Create ASTRA algorithm object on GPU."""
+
+    direction = 'forward'
+    ndim = 2
+    vol_id = odl.tomo.astra_data(VOL_GEOM_2D, 'volume', ndim=ndim)
+    rec_id = odl.tomo.astra_data(VOL_GEOM_2D, 'volume', ndim=ndim)
+    sino_id = odl.tomo.astra_data(PROJ_GEOM_2D, 'projection', ndim=ndim)
 
     # 2D CUDA
     proj_id = odl.tomo.astra_projector('nearest', VOL_GEOM_2D, PROJ_GEOM_2D,

